sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala

```diff
@@ -1485,21 +1487,27 @@ class SparkSqlAstBuilder(conf: SQLConf) extends AstBuilder(conf) {
 
       case null =>
         // Use default (serde) format.
-        val name = conf.getConfString("hive.script.serde",
-          "org.apache.hadoop.hive.serde2.lazy.LazySimpleSerDe")
+        val name = if (isInFormat) {
+          conf.getConfString("hive.script.serde",
+            "org.apache.hadoop.hive.serde2.DelimitedJSONSerDe")
```

Comment:
`DelimitedJSONSerDe` doesn't support deserialize.
Use `DelimitedJSONSerDe` for input SerDe and `LazySimpleSerDe` for output SerDe.